src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp

```diff
@@ -308,146 +233,125 @@ namespace OpenMS
+  ) const
+  {
+    Int n_heavy{0}, n_light{0}, n_quant{0}, n_detect{0}, n_ident{0}, n_trans{0};
+    std::map<String, Int> output;
+    for (const Feature& subordinate : component_group.getSubordinates())
     {
-
-      // extract out the matching transition
-      ReactionMonitoringTransition transition;
-      for (size_t trans_it = 0; trans_it < transitions.getTransitions().size(); ++trans_it)
-      {
-        if (transitions.getTransitions()[trans_it].getNativeID() == component_group.getSubordinates()[cg_it].getMetaValue("native_id"))
+      ++n_trans;
+      const String label_type = (String)subordinate.getMetaValue("LabelType");
+      if (label_type == "Heavy") ++n_heavy;
+      else if (label_type == "Light") ++n_light;
+      std::vector<ReactionMonitoringTransition>::const_iterator transition = find_if(
+        transitions.getTransitions().begin(),
+        transitions.getTransitions().end(),
+        [&subordinate](const ReactionMonitoringTransition& tr)
         {
-          transition = transitions.getTransitions()[trans_it];
-          break;
+          return subordinate.getMetaValue("native_id") == tr.getNativeID();
         }
-      }
-
-      // count labels and transition types
-      String label_type = (String)component_group.getSubordinates()[cg_it].getMetaValue("LabelType");
-      if (label_type == "Heavy")
-      { 
-        ++n_heavy;
-      }
-      else if (label_type == "Light")
-      {
-        ++n_light;
-      }
-      if (transition.isQuantifyingTransition())
-      {
-        ++n_quant;
-      }
-      if (transition.isIdentifyingTransition())
-      {
-        ++n_ident;
-      }
-      if (transition.isDetectingTransition())
-      {
-        ++n_detect;
-      }
-      ++n_trans;
+      );
+      if (transition->isQuantifyingTransition()) ++n_quant;
+      if (transition->isIdentifyingTransition()) ++n_ident;
+      if (transition->isDetectingTransition()) ++n_detect;
     }
-
-    // record
     output["n_heavy"] = n_heavy;
     output["n_light"] = n_light;
     output["n_quantifying"] = n_quant;
     output["n_identifying"] = n_ident;
     output["n_detecting"] = n_detect;
     output["n_transitions"] = n_trans;
-
     return output;
   }
-  
-  double MRMFeatureFilter::calculateIonRatio(const Feature & component_1, const Feature & component_2, const String & feature_name)
+
+  double MRMFeatureFilter::calculateIonRatio(
+    const Feature& component_1,
+    const Feature& component_2,
+    const String& feature_name
+  ) const
   {
-    
-    double ratio = 0.0;
+    double ratio{0};
     if (component_1.metaValueExists(feature_name) && component_2.metaValueExists(feature_name))
     {
-      double feature_1 = component_1.getMetaValue(feature_name);
-      double feature_2 = component_2.getMetaValue(feature_name);
-      ratio = feature_1/feature_2;
-      
-    } 
+      ratio = (double)component_1.getMetaValue(feature_name) / (double)component_2.getMetaValue(feature_name);
```

Comment:
I do appreciate the effort, but I hope that you are not spending too much time trying to reduce the number of lines

Reply:
In this case I think it's reasonable to go back to the previous version.
Does it make sense to rename `feature_x` to `ion_x`? Or is it common to identify a feature by its ion's value?